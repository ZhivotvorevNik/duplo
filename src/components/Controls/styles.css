.controls {
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    cursor: pointer;

    &__bottom {
        position: absolute;
        width: calc(100% - 10px);
        bottom: 0;
        display: flex;
        padding: 5px;
        margin: 5px;
        box-sizing: border-box;
        background: rgba(0, 0, 0, 0.5);
        border-radius: 5px;
        transition: visibility 0s, opacity 0.5s linear;
        visibility: visible;
        opacity: 1;
        cursor: auto;
    }

    &_hidden {
        cursor: none;
    }

    &_hidden &__bottom {
        transition: visibility 0.2s, opacity 0.2s linear;
        visibility: hidden;
        opacity: 0;
        pointer-events: none;
    }
}