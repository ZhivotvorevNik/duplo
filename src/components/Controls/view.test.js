import React from 'react';
import Controls from './index.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <Controls />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});