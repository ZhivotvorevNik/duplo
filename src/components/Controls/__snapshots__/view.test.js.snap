// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="controls controls_hidden"
  onClick={undefined}
  onMouseMove={[Function]}
>
  <div
    className="play-button"
    role="button"
  >
    <div
      className="play-button__inner"
    />
  </div>
  <div
    className="controls__bottom"
    onClick={[Function]}
  >
    <svg
      className="mute-button"
      height="24"
      onClick={undefined}
      viewBox="0 0 75 75"
      width="24"
    >
      <g
        stroke="#fff"
        strokeWidth="5"
      >
        <path
          d="M39.389 13.769L22.235 28.606H6v19.093h15.989l17.4 15.051V13.769z"
          fill="#fff"
          strokeLinejoin="round"
        />
        <path
          d="M48.128 49.03a20.087 20.087 0 0 0-.085-21.453M55.082 20.537a29.86 29.86 0 0 1 5.884 17.84 29.83 29.83 0 0 1-5.788 17.699M61.71 62.611a38.952 38.952 0 0 0 8.418-24.233 38.968 38.968 0 0 0-8.519-24.368"
          fill="none"
          strokeLinecap="round"
        />
      </g>
    </svg>
    <input
      className="timeline"
      max={undefined}
      min={0}
      onChange={[Function]}
      type="range"
      value={undefined}
    />
    <div
      className="countdown"
    >
      NaN:NaN/NaN:NaN
    </div>
  </div>
</div>
`;
